<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="UCSCXenaTools">
<title>UCSCXenaTools: an R package for Accessing Genomics Data from UCSC Xena platform, from Cancer Multi-omics to Single-cell RNA-seq â€¢ UCSCXenaTools</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="UCSCXenaTools: an R package for Accessing Genomics Data from UCSC Xena platform, from Cancer Multi-omics to Single-cell RNA-seq">
<meta property="og:description" content="UCSCXenaTools">
<meta property="og:image" content="https://docs.ropensci.org/UCSCXenaTools/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="../index.html">UCSCXenaTools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.5.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/USCSXenaTools.html">UCSCXenaTools: an R package for Accessing Genomics Data from UCSC Xena platform, from Cancer Multi-omics to Single-cell RNA-seq</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/UCSCXenaTools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>UCSCXenaTools: an R package for Accessing Genomics Data from UCSC Xena platform, from Cancer Multi-omics to Single-cell RNA-seq</h1>
                        <h4 data-toc-skip class="author">Shixiang Wang<br>
ShanghaiTech University</h4>
            
            <h4 data-toc-skip class="date">2024-03-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/UCSCXenaTools/blob/HEAD/vignettes/USCSXenaTools.Rmd" class="external-link"><code>vignettes/USCSXenaTools.Rmd</code></a></small>
      <div class="d-none name"><code>USCSXenaTools.Rmd</code></div>
    </div>

    
    
<p><strong>UCSCXenaTools</strong> is an R package for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. Public omics data from UCSC Xena are supported through <a href="https://xenabrowser.net/datapages/" class="external-link"><strong>multiple turn-key Xena Hubs</strong></a>, which are a collection of UCSC-hosted public databases such as TCGA, ICGC, TARGET, GTEx, CCLE, and others. Databases are normalized so they can be combined, linked, filtered, explored and downloaded.</p>
<p><strong>Who is the target audience and what are scientific applications of this package?</strong></p>
<ul>
<li>Target Audience: cancer and clinical researchers, bioinformaticians</li>
<li>Applications: genomic and clinical analyses</li>
</ul>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Install stable release from CRAN with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"UCSCXenaTools"</span><span class="op">)</span></span></code></pre></div>
<p>You can also install devel version of <strong>UCSCXenaTools</strong> from github with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("remotes")</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ropensci/UCSCXenaTools"</span><span class="op">)</span></span></code></pre></div>
<p>If you want to build vignette in local, please add two options:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ropensci/UCSCXenaTools"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>The minimum versions to run the vignette is <code>1.2.4</code>. <a href="https://github.com/ropensci/UCSCXenaTools/issues" class="external-link">GitHub Issue</a> is a place for discussing any problem.</p>
</div>
<div class="section level2">
<h2 id="data-hub-list">Data Hub List<a class="anchor" aria-label="anchor" href="#data-hub-list"></a>
</h2>
<p>All datasets are available at <a href="https://xenabrowser.net/datapages/" class="external-link uri">https://xenabrowser.net/datapages/</a>.</p>
<p>Currently, <strong>UCSCXenaTools</strong> supports the following data hubs of UCSC Xena.</p>
<ul>
<li>UCSC Public Hub: <a href="https://ucscpublic.xenahubs.net/" class="external-link uri">https://ucscpublic.xenahubs.net/</a>
</li>
<li>TCGA Hub: <a href="https://tcga.xenahubs.net/" class="external-link uri">https://tcga.xenahubs.net/</a>
</li>
<li>GDC Xena Hub: <a href="https://gdc.xenahubs.net/" class="external-link uri">https://gdc.xenahubs.net/</a>
</li>
<li>ICGC Xena Hub: <a href="https://icgc.xenahubs.net/" class="external-link uri">https://icgc.xenahubs.net/</a>
</li>
<li>Pan-Cancer Atlas Hub: <a href="https://pancanatlas.xenahubs.net/" class="external-link uri">https://pancanatlas.xenahubs.net/</a>
</li>
<li>UCSC Toil RNAseq Recompute Compendium Hub: <a href="https://toil.xenahubs.net/" class="external-link uri">https://toil.xenahubs.net/</a>
</li>
<li>PCAWG Xena Hub: <a href="https://pcawg.xenahubs.net/" class="external-link uri">https://pcawg.xenahubs.net/</a>
</li>
<li>ATAC-seq Hub: <a href="https://atacseq.xenahubs.net/" class="external-link uri">https://atacseq.xenahubs.net/</a>
</li>
<li>Singel Cell Xena Hub: <a href="https://singlecellnew.xenahubs.net/" class="external-link uri">https://singlecellnew.xenahubs.net/</a>
</li>
<li>Kids First Xena Hub: <a href="https://kidsfirst.xenahubs.net/" class="external-link uri">https://kidsfirst.xenahubs.net/</a>
</li>
<li>Treehouse Xena Hub: <a href="https://xena.treehouse.gi.ucsc.edu:443/" class="external-link uri">https://xena.treehouse.gi.ucsc.edu:443/</a>
</li>
</ul>
<p>Users can update dataset list from the newest version of UCSC Xena by hand with <code><a href="../reference/XenaDataUpdate.html">XenaDataUpdate()</a></code> function, followed by restarting R and <code><a href="https://docs.ropensci.org/UCSCXenaTools/">library(UCSCXenaTools)</a></code>.</p>
<p>If any url of data hub is changed or a new data hub is online, please remind me by emailing to <a href="mailto:w_shixiang@163.com" class="email">w_shixiang@163.com</a> or <a href="https://github.com/ropensci/UCSCXenaTools/issues" class="external-link">opening an issue on GitHub</a>.</p>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>Download UCSC Xena datasets and load them into R by <strong>UCSCXenaTools</strong> is a workflow with <code>generate</code>, <code>filter</code>, <code>query</code>, <code>download</code> and <code>prepare</code> 5 steps, which are implemented as <code>XenaGenerate</code>, <code>XenaFilter</code>, <code>XenaQuery</code>, <code>XenaDownload</code> and <code>XenaPrepare</code> functions, respectively. They are very clear and easy to use and combine with other packages like <code>dplyr</code>.</p>
<p>To show the basic usage of <strong>UCSCXenaTools</strong>, we will download clinical data of LUNG, LUAD, LUSC from TCGA (hg19 version) data hub.</p>
<div class="section level3">
<h3 id="xenadata-data-frame">XenaData data.frame<a class="anchor" aria-label="anchor" href="#xenadata-data-frame"></a>
</h3>
<p><strong>UCSCXenaTools</strong> uses a <code>data.frame</code> object (built in package) <code>XenaData</code> to generate an instance of <code>XenaHub</code> class, which records information of all datasets of UCSC Xena Data Hubs.</p>
<p>You can load <code>XenaData</code> after loading <code>UCSCXenaTools</code> into R.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/UCSCXenaTools/">UCSCXenaTools</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; =========================================================================================</span></span>
<span><span class="co">#&gt; UCSCXenaTools version 1.5.0</span></span>
<span><span class="co">#&gt; Project URL: https://github.com/ropensci/UCSCXenaTools</span></span>
<span><span class="co">#&gt; Usages: https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; If you use it in published research, please cite:</span></span>
<span><span class="co">#&gt; Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data</span></span>
<span><span class="co">#&gt;   from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq.</span></span>
<span><span class="co">#&gt;   Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627</span></span>
<span><span class="co">#&gt; =========================================================================================</span></span>
<span><span class="co">#&gt;                               --Enjoy it--</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">XenaData</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">XenaData</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 Ã— 17</span></span></span>
<span><span class="co">#&gt;   XenaHosts XenaHostNames XenaCohorts XenaDatasets SampleCount DataSubtype Label</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> https://â€¦ publicHub     Breast Canâ€¦ ucsfNeve_puâ€¦          51 gene expreâ€¦ Neveâ€¦</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> https://â€¦ publicHub     Breast Canâ€¦ ucsfNeve_puâ€¦          57 phenotype   Phenâ€¦</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> https://â€¦ publicHub     Glioma (Koâ€¦ kotliarov20â€¦         194 copy number Kotlâ€¦</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> https://â€¦ publicHub     Glioma (Koâ€¦ kotliarov20â€¦         194 phenotype   Phenâ€¦</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> https://â€¦ publicHub     Lung Canceâ€¦ weir2007_puâ€¦         383 copy number CGH  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> https://â€¦ publicHub     Lung Canceâ€¦ weir2007_puâ€¦         383 phenotype   Phenâ€¦</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># â„¹ 10 more variables: Type &lt;chr&gt;, AnatomicalOrigin &lt;chr&gt;, SampleType &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Tags &lt;chr&gt;, ProbeMap &lt;chr&gt;, LongTitle &lt;chr&gt;, Citation &lt;chr&gt;, Version &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Unit &lt;chr&gt;, Platform &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h3>
<p>Select datasets.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The options in XenaFilter function support Regular Expression</span></span>
<span><span class="fu"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="va">XenaHostNames</span><span class="op">==</span><span class="st">"tcgaHub"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/XenaFilter.html">XenaFilter</a></span><span class="op">(</span>filterDatasets <span class="op">=</span> <span class="st">"clinical"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/XenaFilter.html">XenaFilter</a></span><span class="op">(</span>filterDatasets <span class="op">=</span> <span class="st">"LUAD|LUSC|LUNG"</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">df_todo</span></span>
<span></span>
<span><span class="va">df_todo</span></span>
<span><span class="co">#&gt; class: XenaHub </span></span>
<span><span class="co">#&gt; hosts():</span></span>
<span><span class="co">#&gt;   https://tcga.xenahubs.net</span></span>
<span><span class="co">#&gt; cohorts() (3 total):</span></span>
<span><span class="co">#&gt;   TCGA Lung Cancer (LUNG)</span></span>
<span><span class="co">#&gt;   TCGA Lung Adenocarcinoma (LUAD)</span></span>
<span><span class="co">#&gt;   TCGA Lung Squamous Cell Carcinoma (LUSC)</span></span>
<span><span class="co">#&gt; datasets() (3 total):</span></span>
<span><span class="co">#&gt;   TCGA.LUNG.sampleMap/LUNG_clinicalMatrix</span></span>
<span><span class="co">#&gt;   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span></span>
<span><span class="co">#&gt;   TCGA.LUSC.sampleMap/LUSC_clinicalMatrix</span></span></code></pre></div>
<p>Sometimes we only know some keywords, <code><a href="../reference/XenaScan.html">XenaScan()</a></code> can be used to scan all rows to detect if the keywords exist in <code>XenaData</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x1</span> <span class="op">=</span> <span class="fu"><a href="../reference/XenaScan.html">XenaScan</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">'Blood'</span><span class="op">)</span></span>
<span><span class="va">x2</span> <span class="op">=</span> <span class="fu"><a href="../reference/XenaScan.html">XenaScan</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">'LUNG'</span>, ignore.case <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x1</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: XenaHub </span></span>
<span><span class="co">#&gt; hosts():</span></span>
<span><span class="co">#&gt;   https://ucscpublic.xenahubs.net</span></span>
<span><span class="co">#&gt;   https://tcga.xenahubs.net</span></span>
<span><span class="co">#&gt; cohorts() (6 total):</span></span>
<span><span class="co">#&gt;   Connectivity Map</span></span>
<span><span class="co">#&gt;   TARGET Acute Lymphoblastic Leukemia</span></span>
<span><span class="co">#&gt;   Pediatric tumor (Khan)</span></span>
<span><span class="co">#&gt;   Acute lymphoblastic leukemia (Mullighan 2008)</span></span>
<span><span class="co">#&gt;   TCGA Pan-Cancer (PANCAN)</span></span>
<span><span class="co">#&gt;   TCGA Acute Myeloid Leukemia (LAML)</span></span>
<span><span class="co">#&gt; datasets() (28 total):</span></span>
<span><span class="co">#&gt;   cmap/rankMatrix_reverse</span></span>
<span><span class="co">#&gt;   TARGET_ALL/TARGETcnv_genomicMatrix</span></span>
<span><span class="co">#&gt;   TARGET_ALL/TARGETexp_genomicMatrix</span></span>
<span><span class="co">#&gt;   ...</span></span>
<span><span class="co">#&gt;   TCGA.LAML.sampleMap/mutation_wustl</span></span>
<span><span class="co">#&gt;   TCGA.LAML.sampleMap/mutation_wustl_gene</span></span>
<span><span class="va">x2</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: XenaHub </span></span>
<span><span class="co">#&gt; hosts():</span></span>
<span><span class="co">#&gt;   https://tcga.xenahubs.net</span></span>
<span><span class="co">#&gt; cohorts() (1 total):</span></span>
<span><span class="co">#&gt;   TCGA Lung Cancer (LUNG)</span></span>
<span><span class="co">#&gt; datasets() (15 total):</span></span>
<span><span class="co">#&gt;   TCGA.LUNG.sampleMap/HumanMethylation27</span></span>
<span><span class="co">#&gt;   TCGA.LUNG.sampleMap/HumanMethylation450</span></span>
<span><span class="co">#&gt;   TCGA.LUNG.sampleMap/Gistic2_CopyNumber_Gistic2_all_data_by_genes</span></span>
<span><span class="co">#&gt;   ...</span></span>
<span><span class="co">#&gt;   mc3/LUNG_mc3.txt</span></span>
<span><span class="co">#&gt;   mc3_gene_level/LUNG_mc3_gene_level.txt</span></span></code></pre></div>
<p>Query and download.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/XenaQuery.html">XenaQuery</a></span><span class="op">(</span><span class="va">df_todo</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/XenaDownload.html">XenaDownload</a></span><span class="op">(</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">xe_download</span></span>
<span><span class="co">#&gt; This will check url status, please be patient.</span></span>
<span><span class="co">#&gt; All downloaded files will under directory /tmp/RtmpfomlAT.</span></span>
<span><span class="co">#&gt; The 'trans_slash' option is FALSE, keep same directory structure as Xena.</span></span>
<span><span class="co">#&gt; Creating directories for datasets...</span></span>
<span><span class="co">#&gt; Downloading TCGA.LUNG.sampleMap/LUNG_clinicalMatrix</span></span>
<span><span class="co">#&gt; Downloading TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span></span>
<span><span class="co">#&gt; Downloading TCGA.LUSC.sampleMap/LUSC_clinicalMatrix</span></span></code></pre></div>
<p>Prepare data into R for analysis.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cli</span> <span class="op">=</span> <span class="fu"><a href="../reference/XenaPrepare.html">XenaPrepare</a></span><span class="op">(</span><span class="va">xe_download</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">cli</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "list"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cli</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "LUNG_clinicalMatrix" "LUAD_clinicalMatrix" "LUSC_clinicalMatrix"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="browse-datasets">Browse datasets<a class="anchor" aria-label="anchor" href="#browse-datasets"></a>
</h3>
<p>Create two XenaHub objects:</p>
<ul>
<li>
<code>to_browse</code> - a XenaHub object containing a cohort and a dataset.</li>
<li>
<code>to_browse2</code> - a XenaHub object containing 2 cohorts and 2 datasets.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="va">XenaHostNames</span><span class="op">==</span><span class="st">"tcgaHub"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/XenaFilter.html">XenaFilter</a></span><span class="op">(</span>filterDatasets <span class="op">=</span> <span class="st">"clinical"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/XenaFilter.html">XenaFilter</a></span><span class="op">(</span>filterDatasets <span class="op">=</span> <span class="st">"LUAD"</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">to_browse</span></span>
<span></span>
<span><span class="va">to_browse</span></span>
<span><span class="co">#&gt; class: XenaHub </span></span>
<span><span class="co">#&gt; hosts():</span></span>
<span><span class="co">#&gt;   https://tcga.xenahubs.net</span></span>
<span><span class="co">#&gt; cohorts() (1 total):</span></span>
<span><span class="co">#&gt;   TCGA Lung Adenocarcinoma (LUAD)</span></span>
<span><span class="co">#&gt; datasets() (1 total):</span></span>
<span><span class="co">#&gt;   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span></span>
<span></span>
<span><span class="fu"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="va">XenaHostNames</span><span class="op">==</span><span class="st">"tcgaHub"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/XenaFilter.html">XenaFilter</a></span><span class="op">(</span>filterDatasets <span class="op">=</span> <span class="st">"clinical"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/XenaFilter.html">XenaFilter</a></span><span class="op">(</span>filterDatasets <span class="op">=</span> <span class="st">"LUAD|LUSC"</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">to_browse2</span></span>
<span></span>
<span><span class="va">to_browse2</span></span>
<span><span class="co">#&gt; class: XenaHub </span></span>
<span><span class="co">#&gt; hosts():</span></span>
<span><span class="co">#&gt;   https://tcga.xenahubs.net</span></span>
<span><span class="co">#&gt; cohorts() (2 total):</span></span>
<span><span class="co">#&gt;   TCGA Lung Adenocarcinoma (LUAD)</span></span>
<span><span class="co">#&gt;   TCGA Lung Squamous Cell Carcinoma (LUSC)</span></span>
<span><span class="co">#&gt; datasets() (2 total):</span></span>
<span><span class="co">#&gt;   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix</span></span>
<span><span class="co">#&gt;   TCGA.LUSC.sampleMap/LUSC_clinicalMatrix</span></span></code></pre></div>
<p><code><a href="../reference/XenaBrowse.html">XenaBrowse()</a></code> function can be used to browse dataset/cohort links using your default web browser. At default, this function limits one dataset/cohort for preventing user to open too many links at once.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This will open you web browser</span></span>
<span><span class="fu"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span><span class="op">(</span><span class="va">to_browse</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span><span class="op">(</span><span class="va">to_browse</span>, type <span class="op">=</span> <span class="st">"cohort"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This will throw error</span></span>
<span><span class="fu"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span><span class="op">(</span><span class="va">to_browse2</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in XenaBrowse(to_browse2): This function limite 1 dataset to browse.</span></span>
<span><span class="co">#&gt;  Set multiple to TRUE if you want to browse multiple links.</span></span>
<span></span>
<span><span class="fu"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span><span class="op">(</span><span class="va">to_browse2</span>, type <span class="op">=</span> <span class="st">"cohort"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in XenaBrowse(to_browse2, type = "cohort"): This function limite 1 cohort to browse. </span></span>
<span><span class="co">#&gt;  Set multiple to TRUE if you want to browse multiple links.</span></span></code></pre></div>
<p>When you make sure you want to open multiple links, you can set <code>multiple</code> option to <code>TRUE</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span><span class="op">(</span><span class="va">to_browse2</span>, multiple <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/XenaBrowse.html">XenaBrowse</a></span><span class="op">(</span><span class="va">to_browse2</span>, type <span class="op">=</span> <span class="st">"cohort"</span>, multiple <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="more-usages">More usages<a class="anchor" aria-label="anchor" href="#more-usages"></a>
</h2>
<p>The core functionality has been described above. I write more usages about this package in my website but not here because sometimes package check will fail due to internet problem.</p>
<ul>
<li>
<a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-intro/" class="external-link">Introduction and basic usage of UCSCXenaTools</a> - <a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-intro.pdf" class="external-link">PDF</a>
</li>
<li>
<a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-api/" class="external-link">APIs of UCSCXenaTools</a> - <a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-api.pdf" class="external-link">PDF</a>
</li>
</ul>
<p>Read <a href="https://shixiangwang.github.io/home/en/tools/ucscxenatools-single-gene/" class="external-link">Obtain RNAseq Values for a Specific Gene in Xena Database</a> to see how to get values for single gene. A use case for survival analysis based on single gene expression has been published on rOpenSci, please read <a href="https://ropensci.org/technotes/2019/09/06/ucscxenatools-surv/" class="external-link">UCSCXenaTools: Retrieve Gene Expression and Clinical Information from UCSC Xena for Survival Analysis</a>.</p>
</div>
<div class="section level2">
<h2 id="qa">QA<a class="anchor" aria-label="anchor" href="#qa"></a>
</h2>
<div class="section level3">
<h3 id="how-to-resume-file-from-breakpoint">How to resume file from breakpoint<a class="anchor" aria-label="anchor" href="#how-to-resume-file-from-breakpoint"></a>
</h3>
<p>Thanks to the UCSC Xena team, the new feature â€˜resume from breakpointâ€™ is added and can be done by <strong>XenaDownload()</strong> with the <code>method</code> and <code>extra</code> flags specified.</p>
<p>Of note, the corresponding <code>wget</code> or <code>curl</code> command must be installed by your OS and can be found by R.</p>
<p>The folliwng code gives a test example, the data can be viewed on <a href="https://xenabrowser.net/datapages/?dataset=TcgaTargetGtex_expected_count&amp;host=https%3A%2F%2Ftoil.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443" class="external-link">web page</a>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/UCSCXenaTools/">UCSCXenaTools</a></span><span class="op">)</span></span>
<span><span class="va">xe</span> <span class="op">=</span> <span class="fu"><a href="../reference/XenaGenerate.html">XenaGenerate</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="va">XenaDatasets</span> <span class="op">==</span> <span class="st">"TcgaTargetGtex_expected_count"</span><span class="op">)</span></span>
<span><span class="va">xe</span></span>
<span><span class="va">xq</span> <span class="op">=</span> <span class="fu"><a href="../reference/XenaQuery.html">XenaQuery</a></span><span class="op">(</span><span class="va">xe</span><span class="op">)</span></span>
<span><span class="co"># You cannot resume from breakpoint in default mode</span></span>
<span><span class="fu"><a href="../reference/XenaDownload.html">XenaDownload</a></span><span class="op">(</span><span class="va">xq</span>, destdir <span class="op">=</span> <span class="st">"~/test/"</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># You can do it with 'curl' command</span></span>
<span><span class="fu"><a href="../reference/XenaDownload.html">XenaDownload</a></span><span class="op">(</span><span class="va">xq</span>, destdir <span class="op">=</span> <span class="st">"~/test/"</span>, method <span class="op">=</span> <span class="st">"curl"</span>, extra <span class="op">=</span> <span class="st">"-C -"</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># You can do it with 'wget' command</span></span>
<span><span class="fu"><a href="../reference/XenaDownload.html">XenaDownload</a></span><span class="op">(</span><span class="va">xq</span>, destdir <span class="op">=</span> <span class="st">"~/test/"</span>, method <span class="op">=</span> <span class="st">"wget"</span>, extra <span class="op">=</span> <span class="st">"-c"</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Cite me by the following paper.</p>
<pre><code>Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data
  from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. 
  Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627

# For BibTex
  
@article{Wang2019UCSCXenaTools,
    journal = {Journal of Open Source Software},
    doi = {10.21105/joss.01627},
    issn = {2475-9066},
    number = {40},
    publisher = {The Open Journal},
    title = {The UCSCXenaTools R package: a toolkit for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq},
    url = {http://dx.doi.org/10.21105/joss.01627},
    volume = {4},
    author = {Wang, Shixiang and Liu, Xuesong},
    pages = {1627},
    date = {2019-08-05},
    year = {2019},
    month = {8},
    day = {5},
}</code></pre>
<p>Cite UCSC Xena by the following paper.</p>
<pre><code>Goldman, Mary, et al. "The UCSC Xena Platform for cancer genomics data 
    visualization and interpretation." BioRxiv (2019): 326470.</code></pre>
</div>
<div class="section level2">
<h2 id="acknowledgments">Acknowledgments<a class="anchor" aria-label="anchor" href="#acknowledgments"></a>
</h2>
<p>This package is based on <a href="https://github.com/mtmorgan/XenaR" class="external-link">XenaR</a>, thanks <a href="https://github.com/mtmorgan" class="external-link">Martin Morgan</a> for his work.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
